v2/internal/frontend/assetserver/common.go

```diff
@@ -8,9 +8,26 @@ import (
 	"net/http"
 	"strings"
 
+	"github.com/wailsapp/wails/v2/pkg/options"
+	"github.com/wailsapp/wails/v2/pkg/options/assetserver"
 	"golang.org/x/net/html"
 )
 
+func BuildAssetServerConfig(options *options.App) *assetserver.Options {
+	if opts := options.AssetServer; opts != nil {
+		if options.Assets != nil || options.AssetsHandler != nil {
+			panic("It's not possible to use the deprecated Assets and AssetsHandler options and the new AssetServer option at the same time. Please migrate all your Assets options to the AssetServer option.")
+		}
+
+		return opts
+	}
+
+	return &assetserver.Options{
```

Comment:
Considering this is a small struct and it's not getting called a whole heap of times, I'm wondering if it's worth changing the signatures to accept a struct rather than a struct pointer? At this point in the code, it's gone from being optional to a mandatory struct. Just a thought as it's easier to reason about knowing there won't be nil pointer dereferencing. Thoughts?

Reply:
I think we might change the return value to a struct type instead instead of a struct pointer.

For the input I'm not sure if we can do that, see my reply to your comment on the options struct.

Reply:
I have changed the return value to a struct.